controllers/secretagentconfiguration_controller.go

```diff
@@ -121,12 +119,13 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 					"secret_type", string(key.Type))
 				keyInterface.LoadFromData(secObject.Data)
 			}
-
+			// TODO maybe we don't need to use the slice?
 			// Load key references and data
-			var keyRefSecrets []map[string][]byte
+			var keyRefSecrets map[string][]byte
```

Comment:
You need to allocate the map before assigning values to it
```suggestion
			keyRefSecrets := make(map[string][]byte)
```

Reply:
doh